source/platforms/snowflake.md

```diff
@@ -0,0 +1,343 @@
+      resources:
+        requests:
+          nvidia.com/gpu: 1
+        limits:
+          nvidia.com/gpu: 1
+  endpoints:
+    - name: jupyter
+      port: 8888
+      public: true
+    - name: dask-client
+      port: 8786
+      protocol: TCP
+      public: true
+    - name: dask-dashboard
+      port: 8787
+      public: true
+  volumes:
+    - name: rapids-notebooks
+      source: "@volumes/rapids-notebooks"
+      uid: 1000
+      gid: 1000
+```
+
+Notice that in we mounted the `@volumes/rapids-notebooks` internal stage location to our `/home/rapids/workspace` directory inside of our running container. If you create the `workspace` directory when the container is up, what you save there will persist.
+
+We use `snow-cli` to push this `yaml` file:
+
+```bash
+snow stage copy rapids-snowpark.yaml @specs --overwrite --connection CONTAINER_HOL
+```
+
+Verify that your `yaml` was pushed properly by running the following SQL in the
+Snowflake worksheet:
+
+```sql
+USE ROLE CONTAINER_USER_ROLE;
+LS @CONTAINER_HOL_DB.PUBLIC.SPECS;
+```
+
+## Create and Test the Service
+
+Now that we have successfully pushed the image and the spec YAML, we have all
+the components in Snowflake to create our service. We only need a service name,
+a compute pool and the spec file. Run this SQL in the Snowflake worksheet:
+
+```sql
+USE ROLE CONTAINER_USER_ROLE;
+CREATE SERVICE CONTAINER_HOL_DB.PUBLIC.rapids_snowpark_service
+    in compute pool CONTAINER_HOL_POOL
+    from @specs
+    specification_file='rapids-snowpark.yaml'
+    external_access_integrations = (ALLOW_ALL_EAI);
+```
+
+Run the following to verify that the service is successfully running.
+
+```{code-block} sql
+:force:
+
+CALL SYSTEM$GET_SERVICE_STATUS('CONTAINER_HOL_DB.PUBLIC.rapids_snowpark_service');
+```
+
+Since we specified the `jupyter` endpoint to be public, Snowflake will generate
+a url that can be used to access the service via the browser. To get the url,
+run in the SQL snowflake worksheet:
+
+```sql
+SHOW ENDPOINTS IN SERVICE RAPIDS_SNOWPARK_SERVICE;
+```
+
+Copy the jupyter `ingress_url` in the browser. You will see a jupyter lab with a set of
+notebooks to get started with RAPIDS.
+
+```{figure} /images/snowflake_jupyter.png
+---
+alt: Screenshot of Jupyter Lab with rapids example notebooks directories.
+---
+```
+
+## Shutdown and Cleanup
+
+If you no longer need the service and the compute pool up and running, we can
+stop the service and suspend the compute pool to avoid incurring in any charges.
+
+In the Snowflake SQL worksheet run:
+
+```sql
+USE ROLE CONTAINER_USER_ROLE;
+ALTER COMPUTE POOL CONTAINER_HOL_POOL STOP ALL;
+ALTER COMPUTE POOL CONTAINER_HOL_POOL SUSPEND;
+```
+
+If you want to cleanup completely and remove all of the objects created, run the
+following:
+
+```sql
+USE ROLE CONTAINER_USER_ROLE;
+ALTER COMPUTE POOL CONTAINER_HOL_POOL STOP ALL;
+ALTER COMPUTE POOL CONTAINER_HOL_POOL SUSPEND;
+
+DROP SERVICE CONTAINER_HOL_DB.PUBLIC.RAPIDS_SNOWPARK_SERVICE;
```

Comment:
This line didn't work for me, it just said the service didn't exist, but the rest of the commands did so I think I cleaned up successfully.

Reply:
Interesting, this is as suggested by snowflake. Did your service had a different name or something? I'll give this another try after again soon and check.

Reply:
I am still seeing this

<img width="1136" alt="image" src="https://github.com/user-attachments/assets/d0ebdc4f-c149-41df-8448-7882082adb3a" />